tests/client/test_workspaces.py

```diff
@@ -177,3 +177,43 @@ def test_init_with_missing_workspace(owner: "ServerUser"):
 
     with pytest.raises(ValueError):
         rg.init(api_key=owner.api_key, workspace="missing-workspace")
+
+
+@pytest.mark.asyncio
+async def test_delete_workspace(owner: "ServerUser"):
+    workspace = await WorkspaceFactory.create(name="test_workspace")
+
+    init(api_key=owner.api_key)
+
+    ws = Workspace.from_id(workspace.id)
+    ws.delete()
+
+    with pytest.raises(ValueError):
```

Comment:
We should add the `match` arg here to make sure we're handling the correct exception